src/plasma/plasma_manager.cc

```diff
@@ -897,8 +892,7 @@ void process_data_request(event_loop *loop,
                           int64_t data_size,
                           int64_t metadata_size,
                           ClientConnection *conn) {
-  PlasmaRequestBuffer *buf =
-      (PlasmaRequestBuffer *) malloc(sizeof(PlasmaRequestBuffer));
+  PlasmaRequestBuffer *buf = new PlasmaRequestBuffer();
```

Comment:
Looks like we introduced a bug here. There were multiple instances where we needed to replace `free` -> `delete`, and we didn't replace all of them.

Reply:
Good catch. I doubled checked the code and it looks like the bugs were fixed.

Reply:
Yeah, I think I got most of them. It's possible that I missed some. But now that our valgrind tests are passing again, any problems should show up there.